app/src/components/CommentForm.js

```diff
@@ -1,40 +1,92 @@
 import React from 'react';
-import TextField from '@material-ui/core/TextField';
-import FormControl from '@material-ui/core/FormControl';
+import { withStyles } from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import { compose, withHandlers, withState } from 'recompose';
+import FormControl from './FormControl';
+import errorForField from '../utils/errorForField';
+import validateMessage from '../validators/validateMessage';
 
-const handleSubmit = ({ commentMutation, message }) => () => (
+const styles = theme => ({
+  button: {
+    float: 'right',
+    marginTop: theme.spacing.unit
+  },
+  formLoading: {
+    visibility: 'hidden'
+  },
+  form: {
+    maxWidth: 400,
+    width: '80%',
+    marginTop: theme.spacing.unit * 8,
+    marginBottom: theme.spacing.unit * 2,
+  }
+});
+
+const handleSubmit = ({ commentMutation, message, setErrors, setMessage }) => async event => {
+  event.preventDefault();
+  const errors = validateMessage({ message });
+  setErrors(errors);
+  if (errors) { return; }
   commentMutation({ message })
-);
+    .then(() => {
+      setMessage('');
+    });
+};
 
-const onChangeMessage = ({ setMessage }) => ({ target: { value }}) => {
+const onKeyPress = ({ handleSubmit }) => event => {
+  if (event.key === 'Enter' && event.ctrlKey) {
+    handleSubmit(event);
+  }
+};
+
+const onChangeMessage = ({ errors, setErrors, setMessage }) => ({ target: { value }}) => {
+  if (errorForField(errors, 'message')) {
+    setErrors(null);
+  }
   setMessage(value);
 };
 
 const CommentForm = ({
+  classes,
+  errors,
   handleSubmit,
   message,
   onChangeMessage,
+  onKeyPress,
 }) => (
-  <div>
-    <FormControl fullWidth>
-      <TextField
-        label="Message:"
-        onChange={onChangeMessage}
-        type="message"
-        value={message}
-      />
-    </FormControl>
-    <Button color="primary" onClick={handleSubmit}>Post</Button>
-  </div>
+  <form className={classes.form} onSubmit={handleSubmit}>
+    <FormControl
+      errors={errors}
+      field="message"
+      label="How will you make our world a better place?"
+      onChange={onChangeMessage}
+      value={message}
+      textFieldOptions={{
```

Comment:
pass miscellaneous options to `TextField`, these get spread whereas we want to be explicit with required fields